tensorflow_probability/python/distributions/gev.py

```diff
@@ -0,0 +1,265 @@
+          parameters=parameters,
+          name=name)
+
+  @staticmethod
+  def _param_shapes(sample_shape):
+    return dict(
+        zip(('loc', 'scale', 'concentration'),
+            ([tf.convert_to_tensor(sample_shape, dtype=tf.int32)] * 3)))
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(loc=0, scale=0, concentration=0)
+
+  @property
+  def loc(self):
+    """Distribution parameter for the location."""
+    return self._gev_bijector.loc
+
+  @property
+  def scale(self):
+    """Distribution parameter for scale."""
+    return self._gev_bijector.scale
+
+  @property
+  def concentration(self):
+    """Distribution parameter for shape."""
+    return self._gev_bijector.concentration
+
+  def _entropy(self):
+    # Use broadcasting rules to calculate the full broadcast sigma.
+    scale = self.scale * tf.ones_like(self.loc)
+    return 1. + tf.math.log(scale) + np.euler_gamma*(1. + self.concentration)
+
+  def _log_prob(self, x):
+    with tf.control_dependencies(self._gev_bijector._maybe_assert_valid_x(x)):
+      scale = tf.convert_to_tensor(self.scale)
+      z = (x - self.loc) / scale
+
+      conc = tf.convert_to_tensor(self.concentration)
+      equal_zero = tf.equal(conc, 0.)
+      log_t = tf.where(equal_zero, -z,
+                       -tf.math.log1p(z * conc) / conc)
+
+      return (conc + 1) * log_t - tf.exp(log_t) - tf.math.log(scale)
+
+  def _mean(self):
+    conc = tf.convert_to_tensor(self.concentration)
+    equal_zero = tf.equal(conc, 0.)
+    less_than_one = tf.less(conc, 1.)
+
+    mean_z = tf.where(equal_zero,
+                      np.euler_gamma*tf.ones_like(conc),
+                      tf.where(less_than_one,
+                               tf.math.expm1(tf.math.lgamma(1. - conc)) / conc,
+                               np.inf*tf.ones_like(conc))
+                      )
+
+    return self.loc + self.scale * mean_z
+
+  def _stddev(self):
+    conc = tf.convert_to_tensor(self.concentration)
+    equal_zero = tf.equal(conc, 0.)
+    less_than_half = tf.less(conc, 0.5)
+
+    g1_square = tf.exp(tf.math.lgamma(1. - conc)) ** 2
+    g2 = tf.exp(tf.math.lgamma(1. - 2.*conc))
+
+    std_z = tf.where(equal_zero,
+                     tf.ones_like(conc) * np.pi / np.sqrt(6),
+                     tf.where(less_than_half,
+                              tf.math.sqrt(g2 - g1_square) / tf.abs(conc),
+                              np.inf*tf.ones_like(conc))
+                     )
+
+    return self.scale * tf.ones_like(self.loc) * std_z
+
+  def _mode(self):
+    conc = tf.convert_to_tensor(self.concentration)
+    equal_zero = tf.equal(conc, 0.)
+
+    # Use broadcasting rules to calculate the full broadcast sigma.
+    mode_z = tf.where(equal_zero,
+                      tf.zeros_like(conc),
+                      tf.math.expm1(-conc * tf.math.log1p(conc)) /conc)
+
+    return self.loc + self.scale * mode_z
+
+  def _default_event_space_bijector(self):
```

Comment:
Drop this entirely, the superclass has a good default implementation for this distribution (it'll use the composition of sigmoid + GevCDF). Identity is particularly bad, since this distribution often isn't supported on the entire real line.